frontend/src/views/Crops/Crops.tsx

```diff
@@ -415,254 +437,393 @@ function Crops() {
+      valueGetter: (_value, row) => row?.Crops[0]?.cropName,
+      width: 150,
+      minWidth: 150,
+      maxWidth: 300,
+      sortable: false,
+    },
+    {
+      field: 'cropTypeName',
+      headerName: 'Crop Type',
+      valueGetter: (_value, row) => row?.Crops[0]?.cropTypeName,
+      width: 120,
+      minWidth: 100,
+      maxWidth: 300,
+    },
+    {
+      field: '',
+      headerName: 'Actions',
+      width: 150,
+      renderCell: (cell: any) => {
+        const isRowHasSoilTest = Object.keys(fields[cell?.row?.id]?.Crops)?.length;
+
+        const handleEditRowBtnClick = () => {
+          handleEditCrop(parseInt(cell?.row?.id, 10));
+          setIsDialogOpen(true);
+        };
+        const handleDeleteRowBtnClick = () => {
+          handleDeleteCrop(parseInt(cell?.row?.id, 10));
+        };
+        return (
+          <div>
+            {isRowHasSoilTest ? (
+              <div>
+                <FontAwesomeIcon
+                  css={tableActionButtonCss}
+                  onClick={handleEditRowBtnClick}
+                  icon={faEdit}
                 />
-              </ListItem>
+                <FontAwesomeIcon
+                  css={tableActionButtonCss}
+                  onClick={handleDeleteRowBtnClick}
+                  icon={faTrash}
+                />
+              </div>
             ) : (
-              <>
-                <ListItem>{field.Crops[0].cropName}</ListItem>
-                <ListItem align="right">
-                  <button
-                    type="button"
-                    onClick={() => handleEditCrop(index)}
-                    aria-label={`Edit Crop ${field.Crops[0].cropName}`}
-                  >
-                    <FontAwesomeIcon icon={faEdit} />
-                  </button>
-                  <button
-                    type="button"
-                    onClick={() => handleDeleteCrop(index)}
-                    aria-label={`Delete Crop ${field.Crops[0].cropName}`}
-                  >
-                    <FontAwesomeIcon icon={faTrash} />
-                  </button>
-                </ListItem>
-              </>
+              <ButtonGov
+                variant="primary"
+                size="small"
+                onPress={handleEditRowBtnClick}
+              >
+                Add crop
+              </ButtonGov>
             )}
-          </ListItemContainer>
-        ))}
-      </ContentWrapper>
-
-      {/* Crop edit modal */}
-      {isModalVisible && (
-        <Modal
-          isVisible={isModalVisible}
-          title="Edit Crop"
-          onClose={() => {
-            setIsModalVisible(false);
-            setCalculationsPerformed(false);
-            setErrors({});
-          }}
-          footer={modalFooter}
+          </div>
+        );
+      },
+      sortable: false,
+      resizable: false,
+    },
+  ];
+
+  // Define column headings for Nutrient added/removed tables
+  const requireAndRemoveColumns: GridColDef[] = useMemo(
```

Comment:
I could be wrong, but I don't see this using any variables. I'd recommend throwing this in a constants file

Reply:
I've moved it out of the component itself, but in the same file since it isn't used in any other files. Removed useMemo.